cogs/assessments.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import re
+import arrow
+import discord
+import requests
+from discord.ext import commands
+import time
+import sys
+from datetime import datetime
+
+assessments = []
+
+
+class Assessments(commands.Cog):
+    def __init__(self,bot):
+        self.bot = bot
+        self.load_assessments_file()
+
+    def load_assessments_file(self):
+        #Load file which contains all upcoming assessments
+        if os.path.isfile("assessments.txt"):
+            with open("assessments.txt", "r") as f:
+                for line in f:
+                    line = line.strip()
+                    assessments.append(line)
+        else:
+            open('assessments.txt', 'a').close()
+
+    def update_assessments_file(self):
+        #Update file which contains all upcoming assessments
+        with open("assessments.txt", "w") as f:
+            for assessment in assessments:
+                f.write(assessment)
+                f.write("\n")
+
+    def desc_formatter(self):
+        #Description formatting
+        formatted = []
+        for line in assessments:
+            line = line.strip().split()
+            formatted.append(f"{line[0]} {line[1]} - {' '.join(line[2:])}\n")
+        return formatted
+
+    def date_validator(self, date):
+        #validates a date
+        try:
+            datetime.strptime(date.strip(), "%d/%m/%y %H:%M")
+            #datetime(year=int(year),month=int(month),day=int(day),hour=int(hour), minute=int(minute))
+        except ValueError:
+            return False
+        return True
+
+    def date_sorter(self):
+        #sorts asssessments by dates and times
+        assessments.sort(key=lambda date: datetime.strptime(date[0:14], "%d/%m/%y %H:%M"))
+
+    def remove_entry(self, entry):
+        #removes entry at index entry
+        del assessments[entry]
+        self.update_assessments_file()
+
+    @commands.command(aliases=["CA", "assessments"])
+    async def ca(self,ctx):
+        if len(assessments) > 0:
+            formatted = self.desc_formatter()
+            embed = discord.Embed(title="{}".format("**Upcoming Assessment(s)**"),color=0x78ff83, description="\n".join(formatted))
+            await ctx.send(embed=embed)
+        else:
+            embed = discord.Embed(title="{}".format("**Upcoming Assessment(s)**"),color=0x78ff83, description="You are free....for now >:)")
+            await ctx.send(embed=embed)
+
+    @commands.has_any_role("OVERLORDS", "Mahmood", "Class Rep")
+    @commands.command(aliases=["addCA"])
+    async def addca(self,ctx,date,time,module,*,description):
```

Comment:
this is a perferance format, but I prefer to space the arguments like so `addca(self, ctx, date, time, module, *, description)`